exercises/nucleotide-count/canonical-data.json

```diff
@@ -17,7 +17,30 @@
           }
         },
         {
-          "description": "strand with repeated nucleotide",
+          "comments" : ["Should check for null-references and assume an empty input."],
+          "description": "counts an undefined DNA strand as empty strand.",
+          "property": "nucleotideCounts",
+          "strand": null,
+          "expected": {
+            "A": 0,
+            "C": 0,
+            "G": 0,
+            "T": 0
+          }
+        },
+        {
+          "description": "counts a single nucleotide correctly.",
+          "property": "nucleotideCounts",
+          "strand": "G",
+          "expected": {
+            "A": 0,
+            "C": 0,
+            "G": 1,
+            "T": 0
+          }
```

Comment:
even tough this case might already be included in the next case, this test is for a nice sliming progression. Developers should have quick success by implementing a subset of the spec first.

Reply:
Why is returning a hardcoded 1 different from returning a hardcoded 7?

Reply:
This is a good point. 

I agree that this is not necessary. However, why should anyone hardcode two or more different test cases? Programming means using your brain to accomplish a tasks. The TDD cycle is red-green-refactor(?) and sure, you could hardcode the handling of a one-length input and a n-length input to make it green, but then there is still the third phase: To check if there is any need to improve the code. If someone just hardcodes staements to pass several tests, then I dare to say this person might not want to program in the first place.

My intent with the two test cases was, that it is easier to handle a single character input than a sequence. I think the handling of a sequence is an extension of the single char input. Small steps with fast achievements (in form of another green test) seem to be a big motivator to me, especially for inexperienced developers who have to think hard for every step they take.

Reply:
I can't see the code context for this comment any more. So I can't comment further on that.

> The TDD cycle is red-green-refactor(?) and sure, you could hardcode the handling of a one-length input and an n-length input to make it green, but then there is still the third phase: To check if there is any need to improve the code. If someone just hardcodes statements to pass several tests, then I dare to say this person might not want to program in the first place.

I disagree with this assessment. 

If hardcoded values make the tests pass, then hardcode the values. The refactoring that needs to happen is improving the design of the existing code without altering its behaviour. If after "refactoring" your code can handle more cases than it could previously, you've strayed from the path. 

The solution is to add more tests to test the cases that your code does not handle, and then write the code to pass these tests, and then to refactor to improve the design, again without altering its behaviour.

Reply:
@Insti  Oh, the context disappeared even after committing normally. You can see the contest when viewing the commit directly:

Reply:
> Adding a test for a single-character input
> and following up a test for a multi-char input.

This is a good idea.